hack/deploy-karmada-agent.sh

```diff
@@ -53,7 +53,9 @@ MEMBER_CLUSTER_NAME=$4
 source "${REPO_ROOT}"/hack/util.sh
 
 # install agent to member cluster
-CURR_KUBECONFIG=$KUBECONFIG # backup current kubeconfig
+if [ -n "${KUBECONFIG+x}" ];then
```

Comment:
Just a question, does `[ -n "${KUBECONFIG}" ]` not work?

Reply:
> Just a question, does `[ -n "${KUBECONFIG}" ]` not work?

if KUBECONFIG does not define, it will trigger an 'unbound variable' error.

Reply:
Right, 'unbound variable' error will error because of `set -o nounset`.

https://github.com/karmada-io/karmada/blob/25f28f5006c6afc8a94ed2abebcca28353f7abca/hack/deploy-karmada-agent.sh#L4